src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotPrimaryDataQueryStep.java

```diff
@@ -1,112 +0,0 @@
-      SnapshotRequestModel snapshotReq,
-      AuthenticatedUserRequest userRequest) {
-    this.bigQuerySnapshotPdao = bigQuerySnapshotPdao;
-    this.datasetService = datasetService;
-    this.snapshotService = snapshotService;
-    this.snapshotDao = snapshotDao;
-    this.snapshotReq = snapshotReq;
-    this.userRequest = userRequest;
-  }
-
-  @Override
-  public StepResult doStep(FlightContext context) throws InterruptedException {
-    // TODO: this assumes single-dataset snapshots, will need to add a loop for multiple
-    // (based on the validation flight step that already occurred.)
-    /*
-     * get dataset and assetName
-     * get asset from dataset
-     * which gives the root table
-     * to use in conjunction with the filtered row ids to create this snapshot
-     */
-    Snapshot snapshot = snapshotDao.retrieveSnapshotByName(snapshotReq.getName());
-    SnapshotRequestQueryModel snapshotQuerySpec = snapshotReq.getContents().get(0).getQuerySpec();
-    String snapshotAssetName = snapshotQuerySpec.getAssetName();
-
-    String snapshotQuery = snapshotReq.getContents().get(0).getQuerySpec().getQuery();
-    Query query = Query.parse(snapshotQuery);
-    List<String> datasetNames = query.getDatasetNames();
-    // TODO this makes the assumption that there is only one dataset
-    // (based on the validation flight step that already occurred.)
-    // This will change when more than 1 dataset is allowed
-    String datasetName = datasetNames.get(0);
-
-    Dataset dataset = datasetService.retrieveByName(datasetName);
-    DatasetModel datasetModel = datasetService.retrieveModel(dataset, userRequest);
-
-    // get asset out of dataset
-    Optional<AssetSpecification> assetSpecOp =
-        dataset.getAssetSpecificationByName(snapshotAssetName);
-    AssetSpecification assetSpec =
-        assetSpecOp.orElseThrow(() -> new AssetNotFoundException("Expected asset specification"));
-
-    Map<String, DatasetModel> datasetMap = Collections.singletonMap(datasetName, datasetModel);
-    BigQueryVisitor bqVisitor = new BigQueryVisitor(datasetMap);
-
-    String sqlQuery = query.translateSql(bqVisitor);
-
-    // validate that the root table is actually a table being queried in the query -->
-    // and the grammar only picks up tables names in the from clause (though there may be more than
-    // one)
-    List<String> tableNames = query.getTableNames();
-    String rootTablename = assetSpec.getRootTable().getTable().getName();
-    if (!tableNames.contains(rootTablename)) {
-      throw new InvalidQueryException(
-          "The root table of the selected asset is not present in this query");
-    }
-
-    // now using the query, get the rowIds
-    // insert the rowIds into the snapshot row ids table and then kick off the rest of the
-    // relationship walking
-    Instant createdAt = CommonFlightUtils.getCreatedAt(context);
-
-    bigQuerySnapshotPdao.queryForRowIds(assetSpec, snapshot, sqlQuery, createdAt);
```

Comment:
Note: This "createdAt" concept is something that is not yet implemented for Azure. I believe this is in place to handle the case that we're creating a snapshot at the same time we're performing an ingest. I would have to think through it a bit more, but I actually think that this scenario wouldn't break things for Azure. Since we're creating the parquet files in one transaction with synapse, that parquet file becomes the source of truth for the rest of the snapshot creation process. We're not creating staging tables or anything like that.